### ISSUE TITLE
How do I cite this work?

### ISSUE THREAD

**Issue by cai-lw** [spotify/annoy#299]
I am using this library in my thesis project and I would like to cite it in my thesis. Beside the library itself, I also want to cite some academic stuff.
Are there any academic publication associated with this library? Or do you know about any papers or textbooks that describe your approach of approximate nearest neighbor search?

**Comment by erikbern** [comment]
Sorry, no, I’m an industry practitioner :)

Think you can just include an URL?

**Comment by cai-lw** [comment]
@erikbern I just want to know where the random division tree idea comes from. If you learned about it from a course, a book, or even a blog post, I would like to cite them in my work. If you come up with the idea yourself, then it's fine.

**Comment by erikbern** [comment]
I wrote this library back in 2012, so unfortunately I don't remember, but pretty sure I saw something about it somewhere.

**Comment by eddelbuettel** [comment]
I used the following which produces something closer to what the R Journal,  JSS, ... use for CRAN packages:

```
@Manual{Github:annoy,
  author =       {Erik Bernhardsson},
  title =        {Annoy: Approximate Nearest Neighbors in C++/Python},
  year =         2018,
  note =	 {Python package version 1.13.0},
  url =          {https://pypi.org/project/annoy/},
}
```

The publication really is the PyPi upload, so versioning and dates make some sense.  To me at least :grinning:

**Comment by erikbern** [comment]
makes sense – i can put this in the README.md. does that make sense @eddelbuettel @bartolsthoorn @cai-lw ?

**Comment by brianhie** [comment]
@cai-lw here's a STOC paper describing random projection trees: http://cseweb.ucsd.edu/~dasgupta/papers/rptree-stoc.pdf  This could be what you're looking for?

**Comment by serengil** [comment]
Please add the bibtex in the read-me. So, academicians will be able to cite it in same way.
